### ISSUE TITLE
Political statement in last commit

### ISSUE THREAD

**Issue by cobia** [Yaffle/EventSource#199]
Someone is making a political statement in the source code of the last commit.

**Comment by ejdaly** [comment]
Seems this has also been published as a new version (1.0.26) to npm

https://www.npmjs.com/package/event-source-polyfill/v/1.0.26

**Comment by mrgunnar** [comment]
For what it's worth, this is the text in the statement translated using Google Translate:

> February Russia attacked Ukraine The people of Ukraine are universally mobilized and ready to defend their country from enemy invasion. 91% of Ukrainians fully support their President Volodymyr Zelensky and his response to the Russian attack. The whole world condemned the unjustified invasion and decided to impose unprecedented sanctions against Russia. With each new day, they will be felt more and more strongly among civilians. At the same time, the Russian government restricts citizens' access to external information, planting one-sided formulations and versions of what is happening. As a reliable source of information, download the secure Tor Browser: https://www.torproject.org/ And visit: https://www.bbcweb3hytmzhn5d532owbu6oqadra5z3ar726vq5kgwwn6aucdccrad.onion/english Stop this senseless war! Stop war criminal Putin!

**Comment by mrgunnar** [comment]
As for the behavior:

- The message will appear 15 seconds after this library is loaded, using a `setTimeout`
- The message is only displayed to browsers where the resolved timezone is one of the following identifiers:
    - Asia/Anadyr
    - Asia/Barnaul
    - Asia/Chita
    - Asia/Irkutsk
    - Asia/Kamchatka
    - Asia/Khandyga
    - Asia/Krasnoyarsk
    - Asia/Magadan
    - Asia/Novokuznetsk
    - Asia/Novosibirsk
    - Asia/Omsk
    - Asia/Sakhalin
    - Asia/Srednekolymsk
    - Asia/Tomsk
    - Asia/Ust-Nera
    - Asia/Vladivostok
    - Asia/Yakutsk
    - Asia/Yekaterinburg
    - Europe/Astrakhan
    - Europe/Kaliningrad
    - Europe/Kirov
    - Europe/Moscow
    - Europe/Samara
    - Europe/Saratov
    - Europe/Simferopol
    - Europe/Ulyanovsk
    - Europe/Volgograd
    - W-SU
 - The check uses the `Intl.DateTimeFormat` API, which is not available in all browsers and environments. That means this library might now crash in older browsers, even for users not in the above timezones.
 - The message is displayed using an `alert()`, but uses ANSI color/control codes, which doesn't render properly in an `alert()` window
 - After showing the alert, the browser attempts to open `http://www.change.org/NetVoyne` using a `window.open` call. This often gets blocked by popup blocking in browsers, due to it not being triggered by a user action.

Here is a screenshot of how this appears in Firefox:
![Screen Shot 2022-03-17 at 16 00 58](https://user-images.githubusercontent.com/98051007/158908393-b30ce76d-1a3e-40d0-9356-aab95edcf505.png)

Version 1.0.25 and below does not include this statement.

**Comment by ejdaly** [comment]
In fairness to the project owner - he / she is putting their reputation on the line to oppose something which all agree is unjust and needs action (I just don't agree that the action made here is going to get the right outcome). And - as someone who has benefitted from the authors' work and effort - I want to support them (just not this way). So I'm making a small donation to the funds below, as an alternative, and in solidarity with what I believe is the intended outcome - to support Ukraine.

---
Ukrainian Army Fund
https://savelife.in.ua/en/donate/

Red Cross Humanitarian Fund
https://www.icrc.org/en/donate/ukraine

Children's Emergency Fund
https://www.savethechildren.org/us/where-we-work/ukraine

**Comment by Gemorroj** [comment]
@ejdaly in fact you support the war. You're putting gasoline on the fire.

**Comment by mrdunski** [comment]
@Gemorroj 

Let's put this straight. That's your profile's name:
> Gemorroj **[Z]**

It would be valid point that @ejdaly supports the war (along with helping victims of the war) and puts information unrelated to the issue, but that's exactly what you do as well.
Stop trolling.

Anyway the only reason I'm here is because my boss asked me to verify if my project is impacted by this malicious change.
I agree with @ejdaly that it is not a way to go and there are better options to support a right to defend your country, help war victims and hurt the agressor. This exploits the trust we had in the community and in long run it will hit the OS community more than Russians.

**Comment by ejdaly** [comment]
_personal>>>_

The Lenin caricature tried to put words in my mouth above - so to be very clear - I fully **oppose** the war. 

And sure, that's a personal / political view (not a very controversial one).

_<<<personal_

Now, to focus on getting this Github issue resolved, we need to understand that the commit was added by the project author, and therefore it's likely that they view this Github issue as "not a bug".

And the author owns this open source project - they decide on what does / does not go into it.

So my final comment on this issue is simply an appeal to the author to remove the malicious code, and perhaps replace it with a banner / statement at the top of the README.md to raise awareness / provide information / links to relevant services / call to action / whatever they like.

**Comment by Bedivierre** [comment]
Well. I'm russian, have to say it from the start unless anybody begin to yell "HEY WE CAUGHT RUSSIAN SWINE".
About politic issues in code I have to say - stupid. It can do a bit of damage, but in grand scheme of things it's just a little mischief of a children who were told by parents "That boy Vasya is bad, so spit at him. Do not believe? Trust me, grandaunt of my neighbour heard from her old classmate that his older brother three years ago kicked a really good boy Johnny. Johnny just took his pencil and turned over his table, well, maybe told something, but, do Vasya's brother need to slap him just because of this? Johnny just wanted to make a friend..." 
I hope that there is people who have enough imagination to tell that this is just exaggregation, but still, not too far from truth. I do not know anyone who think that this war is good, and I'm living in St. Petersburg, so I do not meet lack of people to hear an opinion from. And I do not believe any news, be it russian nor western, just because this war is an overhyped local conflict over some political interests, doubtfully Ukranian (military NATO bases in Ukraine maybe?). It's cruel to interpret this war like this, but this is truth. In comparison wars in Yugoslavia, Iraq, Libya, Afganistan was much, much more terrifying, but I haven't heard any about any sanctions, reparations or even just "Sorry" addressing people who died and suffered from hunger, injuries or orphaning there. 
I have to mention that Russia started so-called war in country directly bordering it and where conflict is eight years on rails. Can someone tell me what will happens if Mexico/Canada will start civil war for eight years? Of course brave american peacemakers will march and free countries from suffering. And another question, when USA come to abovementioned countries like Iraq, what they forgot there, on the other side of globe? Chemical weapons? Still not found, different from officially confirmed hundred thousand bodies. And after that (oh, I forgot to mention Fatman and Little Boy, yeah) dare to call Russia as aggressor? Where was your protests of 'peacenotwar' at that time? If you think that you was too young, well, wars in Libya/Iran/Afganistan is still fresh in our memory. If that cannot be called as hypocricy I don't know what cold be.
Sorry, that was just screams of inner demons. I have nothing against common Ukranian people (I personally know many, btw), nor european, chinese, japanese, african and even american (you can not believe me but still), and reason of this war lies with politicians, say, 0.001% of country headcount, and not only russians. You can say anything you want, yep, but if everything I told here is not convinced you that your little games is just a mischief then there is no salvation for you.

**Comment by cx405** [comment]
As of the message and issue itself - to stop the war, one shouldn't start them in first place. 
Its straightaway West countries (US+EU), who have armed and set Ukraine on the rails of war against its neighbor. The path of revolutions is UNLAWFUL as they break the law and UNDEMOCRATIC as never does the revolutionary group represent the majority of the population. It is absurdity to allow on group break the law, while expecting the rest to follow it. US has a very long trail of creating and funding such "color revolutions" (very recent attempts in Belarus, Turkey and Kasachstan) as well as funding religious and ethnic extremists to create the de facto action force on the streets. 
Prior to 2007, Ukraine was trading equally with west AND east to OWN profits. If someone has to stop wars, it should be US itself. However, I believe, their elite clans are so blinded by the greed for profits and control, that they have turned their own principles into little more than a toilet paper. Simply stated, if West Ukraine has the right to destroy the country in a quest for independence, then why cant East Ukraine do the same? And if independence is accounted as being above the law, then it would be naive to expect a non-violent resolution.

**Comment by wallindermont** [comment]
@cx405 @Gemorroj  **stop spamming in issues**, russian programmers.

We are not interested in your views on the war, we know that you support the killing of civilians in Ukraine. Your comments reek of hypocrisy and a desire to bring dirty politics into GitHub discussions.

The author tried to make a commit in support of the people of Ukraine.

It may look silly, but that is his position. If you don't like it, you can make a fork without that commit and use it.

And yes, write your colleagues on Habr that a lot of programmers agree with the author's position. If you and your Russian "colleagues" don't agree, you can go FORWARD WITH YOUR WARSHIP.

**Comment by Gemorroj** [comment]
@wallindermont typical Western neo-Nazism. you don't know anything about our countries and you don't know anything about our life, including Ukraine. But I've been there many times and it's not for you murderers to teach us life.

**Comment by Gemorroj** [comment]
@wallindermont вот только не надо тогда заливать про отсутствие нацизма в укре. Ты можешь лить это в ухи "западным партнерам", но не нам, живущим в одном инорм пространстве. Сколько "москоляк" на гиляках повесил уже? Сколько "колораов" сжег? Из вас вырастили нацистов, а теперь, когда фронт переместился с Донбасса западнее, начинаете визжать на западную, ничего не понимающую аудиторию.

**Comment by cx405** [comment]
> @cx405 @Gemorroj **stop spamming in issues**, russian programmers.
> 
> We are not interested in your views on the war, we know that you support the killing of civilians in Ukraine. Your comments reek of hypocrisy and a desire to bring dirty politics into GitHub discussions.
> 
> The author tried to make a commit in support of the people of Ukraine.
> 
> It may look silly, but that is his position. If you don't like it, you can make a fork without that commit and use it.
> 
> And yes, write your colleagues on Habr that a lot of programmers agree with the author's position. If you and your Russian "colleagues" don't agree, you can go FORWARD WITH YOUR WARSHIP.

Thank you! You just proven a ! BOLD ! example of discrimination and political illiteracy! Please try to counter this:

a) The author used his position to spread political misinformation. I have proven above that this isn't war, to which you countered with zero argument. This IS a military violence of course, as a reaction to stop original military violence against ethnic group by West Ukraine, which was funded and created entirely by and in interest of US+EU. But it isnt war. Right now, the citizens of Mariupol mass testify against extreme crimes of .. ukrainian army on video without anonimity, rape, murder, theft, using civs as human shields. If you really want to find the whole picture, nothing prevents you from doing so but your own courage. 

b) This is a PUBLIC project. Politics are KNOWN to split people into opinions and require deep personal research to understand the whole picture. So this is essentially misuse of a project, a sabotage. It would be wise to keep politics out of social interaction entirely, or prepare to deal with consequences otherwise. Its childish to go fishing on a boat and expect to stay dry.

c) By discriminating against Russians, you allow Russians to discriminate against you or paint you as RACIST. Be prepared for meeting another side of a coin. In addition, you can forget about democracy and liberty - you just bricked it. In other words, you testified that Russians have more liberty than you. Finally, even if you block the truth, you won't be able to run from it the whole life. Its much easier actually to be a man and FACE it. Otherwise, it will hit you hard in the most unexpected moment.

d) The "Russian warship" is a sad testimony of your illiteracy. In reality, Russian battleship captured the Zmeiyniy Island, while its whole Ukrainian crew  ... surrendered and were handled respectfully. But the fun part just begins - Zelensky has expected Ukrainians to die in "honor", declared whole crew as KIA and issued them award of "Hero of Ukraine".
To his surprise, they were all alive and well. Zelensky consequently stripped them of their awards and created this myth about "Russian warship". 

Thus, if you want to make yourself look sillier than you actually are, bring the "Russian warship" argument!

**Comment by cx405** [comment]
> @Gemorroj I know more about Ukraine than you do, it's because I'm Ukrainian. There is no Nazism here, stop listening your TV propaganda.
> 
> By the way, let me remind you official position of Github, Inc:
> 
> ![image](https://user-images.githubusercontent.com/103426477/162981414-8f00c19d-533c-49dc-a291-9b68625827cf.png)
> 
> And if you do not want to be _permanently banned_, you'd better try to choose correct and proper language.

You are in no position to comment on behalf of Github. You just have proven to be biased. That Ukraine is full of Nazi ideology is already proven by West media themself. It is proven by massive amount of captured nationalists with tatooed swastikas, by flags and literature found in Azov and Nationalcorps bases, by your slogans, by "heroes" that you support (Bandera and Shukhetvich), by videos of Fakelmarches and attacks on surviving ww2 heroes on 9 Mai between 2014-2022, by the ethnical crimes you commited in East Ukraine.

Fact, that you are "Ukrainian" changes absolutely nothing in above, as the East Ukraine is also populated by Ukrainians, that resist against you. You are just another person born in Ukraine with Ukrainian pass. Plus in itself the term "Ukrainian" is a geographic and refers to nation. It is NOT ethnic and doesn't refer to nationality.

As a German, I must warn you - if you wont stop spreading misinformation and forcing your political agenda, your actions will have consequences.

**Comment by cx405** [comment]
> 
Das ist richtig, ich verwende die Mail.ru webmail.

**Comment by michael-o** [comment]
> @Gemorroj I know more about Ukraine than you do, it's because I'm Ukrainian. There is no Nazism here, stop listening your TV propaganda.
> 
> By the way, let me remind you official position of Github, Inc:
> 
> ![image](https://user-images.githubusercontent.com/103426477/162981414-8f00c19d-533c-49dc-a291-9b68625827cf.png)
> 
> And if you do not want to be _permanently banned_, you'd better try to choose correct and proper language.

For as long as there is a Stepan Bandera Avenue in Kiev, YES IT IS KIEV, there will be peace in UA. Not even w/o the Russians. GET RID OF YOUR SO CALLED HEROS, they are NOT heros.

**Comment by cx405** [comment]
> I expect consequences from you, tovarisch major. I think the Github team will deal with the 'nazi' words in your comment ;D

Gut so?
https://t.me/stimme_aus_russland/614
https://t.me/stimme_aus_russland/582?single
https://t.me/stimme_aus_russland/141
https://t.me/ponomarb1/12227
https://t.me/ponomarb1/11960
https://t.me/identarist_en/451
https://t.me/identarist_en/441
https://t.me/identarist_en/283
https://t.me/Deutsch_Russische_Freunde/3303
https://t.me/Deutsch_Russische_Freunde/3118
https://t.me/Deutsch_Russische_Freunde/2867
https://t.me/Deutsch_Russische_Freunde/2091

und das mal zum Nachtisch, denn den Frieden wirdst Du mit Deinem Hass und Halbwahrheiten nie zerstören können: https://t.me/Deutsch_Russische_Freunde/1582

**Comment by michael-o** [comment]
> > I expect consequences from you, tovarisch major. I think the Github team will deal with the 'nazi' words in your comment ;D
> 
> Gut so? https://t.me/stimme_aus_russland/614 https://t.me/stimme_aus_russland/582?single https://t.me/stimme_aus_russland/141 https://t.me/ponomarb1/12227 https://t.me/ponomarb1/11960 https://t.me/identarist_en/451 https://t.me/identarist_en/441 https://t.me/identarist_en/283 https://t.me/Deutsch_Russische_Freunde/3303 https://t.me/Deutsch_Russische_Freunde/3118 https://t.me/Deutsch_Russische_Freunde/2867 https://t.me/Deutsch_Russische_Freunde/2091
> 
> und das mal zum Nachtisch, denn den Frieden wirdst Du mit Deinem Hass und Halbwahrheiten nie zerstören können: https://t.me/Deutsch_Russische_Freunde/1582

Das wird kein ukrainischer Politiker zugeben. Dämliche Slogans und der allukrainische Minderwertigkeitskomplex lassen das Land seit 100 Jahren verrotten. Zudem kann wohl die ostslawische Mentalität nicht evolutionieren. Es kommt nicht auf die Größe an, sondern auf die Qualität. Daher zitiere ich Потап и Настя: Всё пучком, если мы не пролезим мы пройдём.

**Comment by quenbyako** [comment]
@cx405 bruh, trust more to russian propaganda. Russian propaganda won't lie to you. Never. Goebbels never lied too. And Simonyan won't do it.

**Comment by Gemorroj** [comment]
@quenbyako лучше сходи к психиатру, ты явно болен

**Comment by cx405** [comment]
> @cx405 bruh, trust more to russian propaganda. Russian propaganda won't lie to you. Never. Goebbels never lied too. And Simonyan won't do it.

Your level of discussion.. this isn't children playground, its github. Where do you get a bright idea that I trust anyone, you included? 

But you still managed to accidentially tuck into a worthy discussion area - the trust. Right now this project is melting down, as the level of trust sunk below the ground all due to the owner action. Who would trust a software, which can purposely start to act in unpredictable way? I believe if owner wanted to send the message, that he is a good person, that rejects any type of violence and prays for his homeland to come out of the deep violent crisis, he would begin many years ago and choose completely different, constructive approach. Maybe if whole Ukraine would reject the way of solving matters with violence, all this wouldn't happen in first place.

**Comment by Bedivierre** [comment]
> @cx405 bruh, trust more to russian propaganda. Russian propaganda won't lie to you. Never. Goebbels never lied too. And Simonyan won't do it.

[https://youtu.be/QA5yZnxS2zc?t=76](url) (timecode 1:16) - Active pro-Ukraine media channel, The Sun. 2M subs, and on video from this link man give his prayers for victims of war in Ukraine's Borodyanka (if you believe description), but this is not important. Important thing is that right behind pope stands a dude with SS Galizien's, aka 14th Waffen SS Grenadier Division "Galicia", mark. Do you need to google what it is? This is not neo-Nazis, this is classic Nazis, truest to their core.

**Comment by michael-o** [comment]
А я трахал Скабееву после эфира.

**Comment by cx405** [comment]
@wallindermont Speaking of hypocrisy, I just noticed the owner of this project uses yandex.ru webmail service.


> The MIT License (MIT)
>
> Copyright (c) 2012 [vic99999@yandex.ru](mailto:vic99999@yandex.ru)

Must be FSB agent. Ridiculous.

**Comment by quenbyako** [comment]
@cx405 
> Maybe if whole **Ukraine** would reject the way of solving matters with violence, all this wouldn't happen in first place.

BRUUUUUUUUUH ![image](https://user-images.githubusercontent.com/13191308/163222415-ac4c8922-3920-4ded-bac3-fc2a0c21388f.png)

This discussion makes no sence, if you don't like what maintainer did in latest commits, do something, report to fss, report to github, report to npm, add this package in your personal blacklist, stop just typing random words all day.

**Comment by michael-o** [comment]
Enjoy the soldier in the center: https://youtu.be/QA5yZnxS2zc?t=41

**Comment by Bedivierre** [comment]
> @Bedivierre died civils are less important than patch on random guy with gun.
> 
> Ok, got it. Thanks. I thought that humanity is imoprtant. Thanks to you, now i understand, that fuck civils, the main goal in our life is to find any nazi symbols and kill those people without asking.
> 
> Thanks bro, you opened my eyes.

@[quenbyako](https://github.com/quenbyako)
Oh. Where was your righteousness when russian soldiers was shot in legs, strangled, tortured (and all of this - on video)? Where was your righteousness when some good doctor from Ukraine on TV for all world told that castrating and killing russians is a good thing? Where was your righteousness when ukrainian soldiers placed their heavy ordnance in civil areas, making this ordnance a legitimate targets for adequate response in war time and turning civilians to cannon fodder simultaneously? Where was your righteousness when west Ukraine forces for eight years bombed Donetsk and Lugansk, cluster bombing included? Where was your humanity while people of those regions live in basements and bomb shelters for months, afraid go out to gather a water to drink? Where was your righteousness when west Ukraine forces thrown Tochka-U missile on Kramatorsk (western media just shut out this news for now, btw)? Don't be hypocrite and open your eyes a bit wider, dude. Read a wiki about SS Galizien and find three differences in methods.

**Comment by Bedivierre** [comment]
> @[michael-o](https://github.com/michael-o)
> 
> > But Zelenskiy told me personally that there are no nazis in UA... :-D
We were told that USSR had no sex too

**Comment by michael-o** [comment]
> > @[michael-o](https://github.com/michael-o)
> > > But Zelenskiy told me personally that there are no nazis in UA... :-D
> 
> We were told that there is no sex in USSR

So there were no gays and people in wheelchairs and no antisemitism. I KNOW all of that shit personally and that SHITTY mentality is still living on in those people.

**Comment by cx405** [comment]
> @cx405
> 
> > Maybe if whole **Ukraine** would reject the way of solving matters with violence, all this wouldn't happen in first place.
> 
> BRUUUUUUUUUH ![image](https://user-images.githubusercontent.com/13191308/163222415-ac4c8922-3920-4ded-bac3-fc2a0c21388f.png)
> 
> This discussion makes no sence, if you don't like what maintainer did in latest commits, do something, report to fss, report to github, report to npm, add this package in your personal blacklist, stop just typing random words all day.


@quenbyako Ukraine ambassador to Germany has openly publicly stated that Ukraine has very large (many thousands) far-radical military force which:

- is tightly integrated into armed forces,
- is controlled by the government,
has played and plays crucial role in current Ukraine

Video: https://cloud.mail.ru/public/BTRT/ZjyjaaxX5

What and where I type is my own personal business.

**Comment by Yaffle** [comment]
Should I print instead something about "Nazi problem"?
